Support returning multiple polygons

orangejulius opened the pull request:
Contains the changes from #10, rebased on top of master to resolve conflicts, and with tests rewritten for clarity.

uatuko committed:
option to limit or return all intersecting polygons

orangejulius committed:
Split test cases into individual tests

For loops in tests quickly get unweildy and hard to read.

trescube commented on index.js:
```diff
@@ -24,20 +24,25 @@ function PolygonLookup( featureCollection ){
   var pt = [ x, y ];
-  for( var ind = 0; ind < bboxes.length; ind++ ){
+  var results = [];
+  for( var ind = 0; ind < bboxes.length && ( limit === undefined || limit === -1 || limit > 0 ); ind++ ){
```
This is really hard to read and leads to some confusing state later.  Could limit be overridden with `MAX_SAFE_INTEGER` when it's initial value is `-1`?

orangejulius replied:
Good idea. I've taken a second to make the logic cleaner, take a look. (it's still not perfect)

orangejulius committed:
Make polygon intersection logic easier to read

orangejulius commented:
Note: I think that while this is fine to merge for now, we should consider a breaking change release where the search function always returns an array of polygon results up to a certain limit. This simplifies the logic a lot. Most places in Pelias probably eventually want to handle multiple point in polygon results anyway.

orangejulius committed:
Add test case for a point that matches two polygons

This is part of the "old behavior" test suite, and it just guarantees
that we didn't break anything when there are multiple matching polygons.

Previously, the first polygon (ordered by id) that matched would be
returned, so this test ensures we have kept that behavior.

orangejulius committed:
Split old and new behavior into separate methods

This will make it easy to remove the old behavior in the future if we'd
like to.

dianashk commented:
LGTM

orangejulius committed:
Add lodash dependency

orangejulius committed:
Use lodash .find to support Node 0.12

Soon we will be able to remove this!

trescube commented on index.js:
```diff
@@ -23,36 +24,94 @@ function PolygonLookup( featureCollection ){
+
+
+/*
+ * Interrnal helper method to return a single matching polygon
```
`Internnal` -> `Internal`

trescube commented:
egregious typo, otherwise :shipit:

dianashk commented:
LGTM
